hbp_validation_framework/__init__.py

```diff
@@ -216,37 +335,608 @@ def get_validation_test(self, test_uri, **params):
+        alias : string
+            User-assigned unique identifier associated with test definition.
+        repository : string
+            URL of Python package repository (e.g. github).
+        path : string
+            Path to test source code within Python package.
+        version : string
+            User-assigned identifier (unique for each test) associated with test instance.
+
+        Returns
+        -------
+        UUID
+            UUID of the test instance that has was edited.
+
+        Examples
+        --------
+        >>> response = test_library.edit_test_instance(test_id="7b63f87b-d709-4194-bae1-15329daf3dec",
+                                        repository="https://github.com/appukuttan-shailesh/morphounit.git",
+                                        path="morphounit.tests.CellDensityTest",
+                                        version="4.0")
+        """
+
+        test_definition_id = test_id    # as needed by API
+        instance_data = locals()
+        for key in ["self", "test_id"]:
+            instance_data.pop(key)
+
+        if instance_id == "" and (test_id == "" or version == "") and (alias == "" or version == ""):
+            raise Exception("instance_id or (test_id, version) or (alias, version) needs to be provided for finding a test instance.")
+        elif instance_id:
+            url = self.url + "/validationtestscode/?id=" + instance_id + "&format=json"
+            #url = self.url + "/validationtestscode/?format=json"
+        elif test_id and version:
+            url = self.url + "/validationtestscode/?test_definition_id=" + test_id + "&version=" + version + "&format=json"
+        else:
+            url = self.url + "/validationtestscode/?test_alias=" + alias + "&version=" + version + "&format=json"
+
+        headers = {'Content-type': 'application/json'}
+        response = requests.put(url, data=json.dumps([instance_data]), auth=self.auth, headers=headers)
+        if str(response) == "<Response [202]>":
+            return response.content
+        else:
+            raise Exception("Error in editing test instance. Response = " + str(response.content))
+
     def _load_reference_data(self, uri):
-        # Load the reference data ("observations")
-        # For now this is assumed to be in JSON format, but we
-        # should support other data formats.
-        # For now, data is assumed to be on the local disk, but we
-        # need to add support for remote downloads.
+        # Load the reference data ("observations"). For now this is assumed
+        # to be in JSON format, but we should support other data formats.
         parse_result = urlparse(uri)
         datastore = URI_SCHEME_MAP[parse_result.scheme](auth=self.auth)
         observation_data = datastore.load_data(uri)
         return observation_data
 
-    def register(self, test_result, project=None, data_store=None):
+    def get_options(self, param=""):
```

Comment:
I think "parameters" is ambiguous here. I suggest using "test attributes", and listing the possible attributes (species, etc.) near the top of the docstring

Reply:
Done.